cmd/lifecycle/analyzer.go

```diff
@@ -20,51 +21,70 @@ import (
 
 type analyzeCmd struct {
 	//flags: inputs
-	cacheDir      string
-	cacheImageTag string
-	groupPath     string
-	uid, gid      int
 	analyzeArgs
+	uid, gid int
+
+	cacheDir      string // Platform API < 0.7
+	cacheImageTag string // Platform API < 0.7
+	groupPath     string // Platform API < 0.7
 
 	//flags: paths to write data
 	analyzedPath string
 }
 
 type analyzeArgs struct {
-	//inputs needed when run by creator
-	imageName  string
-	layersDir  string
-	skipLayers bool
-	useDaemon  bool
-
-	platform cmd.Platform
-
-	//construct if necessary before dropping privileges
-	docker   client.CommonAPIClient
-	keychain authn.Keychain
+	imageName   string
+	layersDir   string
+	orderPath   string //nolint - Platform API >= 0.7
+	runImageRef string //nolint - Platform API >= 0.7
+	stackPath   string //nolint - Platform API >= 0.7
+	skipLayers  bool   // Platform API < 0.7
+	useDaemon   bool
+
+	additionalTags cmd.StringSlice        //nolint Platform API >= 0.7
+	cache          lifecycle.Cache        // Platform API < 0.7
+	docker         client.CommonAPIClient // construct if necessary before dropping privileges
+	group          buildpack.Group        // Platform API < 0.7
+	keychain       authn.Keychain
+	platform       cmd.Platform
 }
 
 func (a *analyzeCmd) DefineFlags() {
 	cmd.FlagAnalyzedPath(&a.analyzedPath)
-	cmd.FlagCacheDir(&a.cacheDir)
-	cmd.FlagCacheImage(&a.cacheImageTag)
-	cmd.FlagGroupPath(&a.groupPath)
+	if a.restoresLayerMetadata() {
+		cmd.FlagCacheImage(&a.cacheImageTag)
+		cmd.FlagCacheDir(&a.cacheDir)
+		cmd.FlagGroupPath(&a.groupPath)
+		cmd.FlagSkipLayers(&a.skipLayers)
+	}
 	cmd.FlagLayersDir(&a.layersDir)
-	cmd.FlagSkipLayers(&a.skipLayers)
+	if a.platformAPIVersionGreaterThan06() {
+		cmd.FlagOrderPath(&a.orderPath)
+		cmd.FlagPreviousImage(&a.imageName)
+		cmd.FlagRunImage(&a.runImageRef)
+		cmd.FlagStackPath(&a.stackPath)
+		cmd.FlagTags(&a.additionalTags)
+	}
 	cmd.FlagUseDaemon(&a.useDaemon)
 	cmd.FlagUID(&a.uid)
 	cmd.FlagGID(&a.gid)
 }
 
 func (a *analyzeCmd) Args(nargs int, args []string) error {
-	if nargs != 1 {
-		return cmd.FailErrCode(fmt.Errorf("received %d arguments, but expected 1", nargs), cmd.CodeInvalidArgs, "parse arguments")
-	}
-	if args[0] == "" {
-		return cmd.FailErrCode(errors.New("image argument is required"), cmd.CodeInvalidArgs, "parse arguments")
+	if !a.platformAPIVersionGreaterThan06() {
```

Comment:
```suggestion
	if !a.supportsImageArgument() {
```

Reply:
Should we warn if the argument is provided for newer platform API?

Reply:
Asked you for guidance here: https://github.com/buildpacks/lifecycle/pull/582#discussion_r616783509. Going to close this conversation as I did the refactoring already.